src/mpi_t/pvar_impl.c

```diff
@@ -7,16 +7,24 @@
 #include "utlist.h"
 
 /* Define storage for the ALL_HANDLES constant */
-MPIR_T_pvar_handle_t MPIR_T_pvar_all_handles_obj = {
```

Comment:
FWIW, I think designated initializers are OK to use now. We use in limited places already. IMO, more usage is good.

Reply:
Okay, I guess I don't disagree. However, here is a bit tricky since the `kind` field need be protected by `#ifdef HAVE_ERROR_CHECKING`. I think the solution of initializing it inside the explicit `init` routine is not a bad choice.